server/service/src/requisition_line/response_requisition_line/delete.rs

```diff
@@ -72,6 +74,16 @@ fn validate(
         return Err(OutError::CannotDeleteLineFromTransferredRequisition);
     }
 
+    let invoice_lines = InvoiceLineRepository::new(connection).query_by_filter(
+        InvoiceLineFilter::new().requisition_id(EqualFilter::equal_to(&requisition_row.id)),
+    )?;
+
+    if invoice_lines.iter().any(|invoice_line| {
+        requisition_line_row.item_link_id == invoice_line.invoice_line_row.item_link_id
+    }) {
+        return Err(OutError::CannotDeleteLineLinkedToShipment);
+    }
```

Comment:
could it... have a test 🥺

Reply:
[e732188](https://github.com/msupply-foundation/open-msupply/pull/6479/commits/e732188b4e94ff37cdafefd582bec9f157fa2cc0)